examples/platforms/posix/ble_nimble.c

```diff
@@ -0,0 +1,1111 @@
+/*
+ *
+ *    Copyright (c) 2018 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    This document is the property of Nest. It is considered
+ *    confidential and proprietary information.
+ *
+ *    This document may not be reproduced or transmitted in any form,
+ *    in whole or in part, without the express written permission of
+ *    Nest.
+ *
+ */
+
+/**
+ *    @file
+ *      This file implements OpenThread BLE platform interface
+ *      that maps into nlplatform port of NimBLE API.
+ *
+ */
+
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <common/logging.hpp>
+#include <openthread/platform/ble.h>
+#include <openthread/platform/ble_hci.h>
+
+#include "ble_nimble_event.h"
+
+#include <nimble/nimble_npl.h>
+#include <nimble/nimble_port.h>
+
+#include "ble_l2cap_priv.h"
+#include <host/ble_hs.h>
+#include <nimble/ble.h>
+#include <services/ans/ble_svc_ans.h>
+#include <services/gap/ble_svc_gap.h>
+#include <services/gatt/ble_svc_gatt.h>
+
+#define TASK_DEFAULT_PRIORITY 1
+#define TASK_DEFAULT_STACK NULL
+#define TASK_DEFAULT_STACK_SIZE 400
+
+/// Max amount of time in [ms] to scan on connection request.
+#define DEFAULT_CONN_DISC_INTERVAL 1000
+#define DEFAULT_ADDR_TYPE BLE_OWN_ADDR_RANDOM
+
+#define BLE_MAX_NUM_SERVICES 2
+#define BLE_MAX_NUM_CHARACTERISTICS 5
+#define BLE_MAX_NUM_UUIDS (BLE_MAX_NUM_SERVICES + BLE_MAX_NUM_CHARACTERISTICS)
+
+bool     sNimbleInitialized = false;
+bool     sNimbleRunning     = false;
+uint16_t sNimbleConnHandle;
+
+static struct ble_npl_task sTaskBleHost;
+static struct ble_npl_task sTaskBleController;
+static struct ble_npl_sem  sTaskBleSyncSem;
+
+// Note: static allocation of GATT database. Tune accordingly.
+ble_uuid_any_t          sNimbleUuids[BLE_MAX_NUM_UUIDS];
+struct ble_gatt_svc_def sNimbleServices[BLE_MAX_NUM_SERVICES];
+struct ble_gatt_chr_def sNimbleCharacteristics[BLE_MAX_NUM_CHARACTERISTICS];
+
+int sNimbleUuidsCount           = 0;
+int sNimbleServicesCount        = 0;
+int sNimbleCharacteristicsCount = 0;
+
+#define L2CAP_COC_MTU (256)
+#define L2CAP_COC_BUF_COUNT (3 * MYNEWT_VAL_BLE_L2CAP_COC_MAX_NUM)
+
+struct ble_l2cap_chan *  sNimbleL2capChannel;
+struct os_mbuf_pool      sNimbleL2capSduMbufPool;
+static struct os_mempool sNimbleL2capSduMemPool;
+static os_membuf_t       sNimbleL2capSduMem[OS_MEMPOOL_SIZE(L2CAP_COC_BUF_COUNT, L2CAP_COC_MTU)];
+
+// NimBLE HCI APIs
+void ble_hci_sock_init(void);
+void ble_hci_sock_set_device(int dev);
+void ble_hci_sock_ack_handler(void *param);
+
+extern uint32_t NODE_ID;
+
+static otError mapNimbleToOtError(int rc)
```

Comment:
nit: I think here and in below functions we should use "a" prefix as code standard suggest.